src/frontend/src/catalog/table_catalog.rs

```diff
@@ -68,12 +68,11 @@ pub struct TableCatalog {
     /// pk_indices of the corresponding materialize operator's output.
     pub stream_key: Vec<usize>,
 
+    pub is_index: bool,
```

Comment:
maybe we can add an enum field here table_type{ Internal, Mview, Index, Source} cc @yezizp2012